talon/signature/extractor.py

```diff
@@ -0,0 +1,195 @@
+"""
+Module with object oriented approach to
```

Comment:
Out of curiosity - why the comment lines are so short? I know shorter lines are easier to read but usually 80 chars is ok. Functions that go on for many lines become hard to read just like lines stretching for many chars.